### PR TITLE
*: cut 2.8.0-rc.0

### PR DESCRIPTION
**Pull request by codesome**
This is an initial draft after 1 parse over the commits.

tsdb changes are taken from the CHANGELOG, but will parse the commits again to see if anything has been missed.

Some PRs needs merging before the release is cut, specifically, I am looking at https://github.com/prometheus/prometheus/pull/5286 (will parse other open PRs soon). 

Also, tsdb needs to be vendored, which will be done soon either by @krasi-georgiev or me.

Will change the state of PR from Draft to Ready when all pending PRs are merged and tsdb is vendored. Anyway, it is good for a review.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,3 +1,16 @@
+## 2.8.0-rc.0 / 2019-03-01
+
+* [CHANGE] `prometheus_tsdb_storage_blocks_bytes_total` is now `prometheus_tsdb_storage_blocks_bytes`. prometheus/tsdb#506
+* [FEATURE] WAL for Remote write API. #4588
```
Comment (brian-brazil) [inline CHANGELOG.md]: ENHANCEMENT, this nominally isn't user visible

**Comment by codesome** [comment]
#5153 was added in 2.7.1

**Comment by codesome** [comment]
Also #5186, in 2.7.1

**Comment by simonpasquier** [comment]
It misses https://github.com/prometheus/prometheus/pull/5189 (enhancement) and https://github.com/prometheus/prometheus/pull/5182 (bug fix + new metrics)

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,3 +1,17 @@
+## 2.8.0-rc.0 / 2019-03-01
+
+* [CHANGE] `prometheus_tsdb_storage_blocks_bytes_total` is now `prometheus_tsdb_storage_blocks_bytes`. prometheus/tsdb#506
+* [FEATURE] Time overlapping blocks are now allowed; vertical compaction and vertical query merge. It is an optional feature which is controlled by `--storage.abc` flag [TODO(codesome): modify the flag after vendoring], disabled by default. prometheus/tsdb#370
+* [ENHANCEMENT] WAL for Remote write API. #4588
```
Comment (tomwilkie) [inline CHANGELOG.md]: `[ENHANCEMENT] Use the WAL for remote_write API.`

**Comment by codesome** [comment]
Waiting on
* https://github.com/prometheus/prometheus/issues/5299 / https://github.com/prometheus/prometheus/pull/5300 (new)
* https://github.com/prometheus/prometheus/pull/5290 (ready)
* https://github.com/prometheus/prometheus/pull/5292 (ready)

File: CHANGELOG.md [hunk 3]
```diff
@@ -1,3 +1,24 @@
+## 2.8.0-rc.0 / 2019-03-06
+
+* [CHANGE] `prometheus_tsdb_storage_blocks_bytes_total` is now `prometheus_tsdb_storage_blocks_bytes`. prometheus/tsdb#506
+* [FEATURE] [EXPERIMENTAL] Time overlapping blocks are now allowed; vertical compaction and vertical query merge. It is an optional feature which is controlled by `--storage.tsdb.allow-overlapping-blocks` flag, disabled by default. prometheus/tsdb#370
```
Comment (juliusv) [inline CHANGELOG.md]: "controlled by" -> "controlled by the"

File: CHANGELOG.md [hunk 4]
```diff
@@ -1,3 +1,24 @@
...
+* [ENHANCEMENT] Use the WAL for remote_write API. #4588
+* [ENHANCEMENT] Query performance improvements. prometheus/tsdb#531
+* [ENHANCEMENT] UI enhancements with upgrade to Bootstrap 4. #5226
+* [ENHANCEMENT] Reduction time that alertmanagers are in flux when reloaded. #5126
```
Comment (juliusv) [inline CHANGELOG.md]: Reduction -> Reduce
alertmanagers -> Alertmanagers

File: CHANGELOG.md [hunk 5]
```diff
@@ -1,3 +1,24 @@
...
+* [ENHANCEMENT] Limit number of metrics displayed on UI to 10000. #5139
+* [ENHANCEMENT] (1) Remember All/Unhealthy choice on target-overview when reloading page. (2) Resize text-input area on Graph page on mouseclick. #5201
+* [ENHANCEMENT] In `histogram_quantile` merge buckets with equivalent le values. #5158.
+* [ENHANCEMENT] Show list of offending labels in the error message in many-to-many scenarios. #5189
+* [BUGFIX] Fix sorting of rule groups. #5260
+* [BUGFIX] discovery/kubernetes: fix support for password_file and bearer_token_file. #5211
```
Comment (juliusv) [inline CHANGELOG.md]: Bit inconsistent to start this with "discovery/kubernetes" (probably just taken from the PR) when all the other changelog files don't start with a prefix like that. IMO better to just reformulate it.

Comment (codesome) [reply CHANGELOG.md]: Now does it look fine?

**Comment by juliusv** [comment]
:+1:

**Comment by krasi-georgiev** [comment]
this runs the Prombench with the changes form the PR and compares with master.
you need to run it and compare with the previous release

**Pull request merged**